lib/compat/wordpress-6.5/fonts/class-wp-font-utils.php

```diff
@@ -20,11 +20,41 @@
 	 * @access private
 	 */
 	class WP_Font_Utils {
+
+		/**
+		 * Adds surrounding quotes to font family names that contain special characters.
+		 *
+		 * It follows the recommendations from the CSS Fonts Module Level 4.
+		 * @link https://www.w3.org/TR/css-fonts-4/#font-family-prop
+		 *
+		 * @since 6.5.0
+		 * @access private
+		 *
+		 * @see sanitize_font_family()
+		 *
+		 * @param string $item A font family name.
+		 * @return string The font family name with surrounding quotes if necessary.
+		 */
+		private static function maybe_add_quotes( $item ) {
+			// Match any non alphabetic characters (a-zA-Z), dashes -, or parenthesis ().
+			$regex = '/[^a-zA-Z\-()]+/';
```

Comment:
I see [in the spec](https://www.w3.org/TR/css-fonts-4/#family-name-syntax), it says

> To avoid mistakes in escaping, it is recommended to quote font family names that contain white space, digits, or punctuation characters other than hyphens

I'm curious, why allow unquoted names with parentheses here, as well?

Reply:
I'm 100% not sure about the '()' characters because the spec has examples like this, but I have never saw outside of those examples.
```
body { font-family: "Adobe Fangsong Std R", generic(fangsong), serif}
```

Reply:
Note that [according to this W3C notice](https://www.w3.org/TR/clreq-gap/#issue587_fonts), no major browsers support font family names like `generic(fangsong)`, yet, (at least for Chinese fonts).

Though I think it would be best for us to anticipate this will be supported sometime in the future.

Reply:
Adding generic support here: https://github.com/WordPress/gutenberg/pull/59103